veps/sig-compute/vmtemplate-proposal.md

```diff
@@ -0,0 +1,733 @@
+
+#### common-templates - Preview branch
+
+The proposed feature will be prototyped by adapting the existing
+`kubevirt/common-templates` project to use the new `VirtualMachineTemplate` CRD,
+demonstrating a clear migration path. A `preview` branch will be cut within the
+common-templates project switching the existing OpenShift `Template` based
+resources over to `VirtualMachineTemplate`. To simplify maintenance of the new
+templates, instance types and preferences shall be used to allow creating
+differently sized VMs from a single template and to reduce the duplication of
+runtime settings.
+
+If this proposal is successful, the `kubevirt/common-templates` project would
+eventually be archived, with its functionality fully superseded by the new
+`virtual-machine-templates` repository and CRDs.
+
+### template.kubevirt.io/v1beta1
+
+#### Extended status in VirtualMachineTemplates
+
+To better keep track of the resources associated with a `VirtualMachineTemplate`
+the status of the CRD could be extended with a `Ready` condition and a list of
+volumes on which the template is dependent. The list should also contain the
+`Ready` condition of the volumes.
+
+This function requires that the `VirtualMachineTemplate` controller evaluates
+all `DataVolumeTemplates` in a specific template and constructs the status from
+all volumes that it can resolve.
+
+#### Import/Export of VirtualMachineTemplates in standardized format
+
+As previously mentioned, to make VM templates reusable, an import and export
+mechanism is required. Ideally, this mechanism should use a common and
+standardized format, to not only allow the import of templates created by
+KubeVirt, but also the import of appliances created by software vendors and
+templates created through other tools. As a widely adopted open standard in the
+virtualization industry, OVA/OVF is the ideal format for KubeVirt to implement.
+
+The entry point for the import mechanism should be a new
+`VirtualMachineTemplateImportRequest `CRD, which contains the required fields to
+specify a remote location where an OVA/OVF file can be obtained from and where
+it should be imported to. Another option could be the direct upload into the
+cluster through `virtctl` and possibly with the help of CDI.
+
+Once the OVA/OVF file can be processed, the contained disks need to be imported
+into PVCs and the attached virtual machine definition needs to be converted into
+a KubeVirt `VirtualMachine`. Once these tasks are completed the controller can
+create a new `VirtualMachineTemplate` from the results. This new template can
+then be used to create actual `VirtualMachines` from the imported template.
+
+Similar to the import mechanism, the entry point to the export mechanism should
+be a new `VirtualMachineTemplateExportRequest` CRD containing the required
+fields to specify either an existing `VirtualMachineTemplate` to be exported or
+an existing `VirtualMachine` to be exported directly. Once the exported template
```

Comment:
I'd avoid extending this to also handle the `VirtualMachine` export case, we already have support for that elsewhere and users can always create a template from the VM first and then export that if they want to.

Reply:
Does it cost much more to automate the workflow from VM -> Template -> Export? IMO it makes for a nicer user experience.